Expensify/App
# Start accounting for notificationPreferences in the unread indicator for Chat Rooms

@jasperhuangg please review

### Details
When users have notificationPreferences in E.Chat for a given, room (comes live with https://github.com/Expensify/Expensify/issues/161785 cc @jasperhuangg) - follow them for push notifications within the app. 

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/161781

### Tests/QA
1. Set your notification preferences for a policy's announce room to `"always"` by running the following JS in E.Web (User 1):
```JD
p = Policy.getCurrent();
announceID = p.policy.chatReportIDAnnounce
API._performPOSTRequest("Report_UpdateNotificationPreference", {
    reportID: announceID,
    notificationPreference: 'always'
});
```
2. For Phones, keep E.Chat running in the background for User 1, for Web/Desktop, have the tab or window open but not focused.
3. Open up E.Chat in another client from a different account that's also in the same policy (User 2)
4. From that account send 4 messages in the #announce room for that policy.
5. Make sure that the app indicator for the client User 1 is in shows `4`

---

1. Set your notification preferences for a policy's announce room to `"daily"` by running the following JS in Web Expensify (OldDot) (User 1):
```JD
p = Policy.getCurrent();
announceID = p.policy.chatReportIDAnnounce
API._performPOSTRequest("Report_UpdateNotificationPreference", {
    reportID: announceID,
    notificationPreference: 'daily'
});
```
2. For Phones, keep E.Chat running in the background for User 1, for Web/Desktop, have the tab or window open but not focused.
3. Open up E.Chat in another client from a different account that's also in the same policy (User 2)
4. From that account send a message in the #announce room for that policy.
5. Make sure that the app indicator for the client User 1 is using shows `1` or is simply blank but present.

---

1. Set your notification preferences for a policy's announce room to `"mute"` by running the following JS in Web Expensify (OldDot) (User 1):
```JD
p = Policy.getCurrent();
announceID = p.policy.chatReportIDAnnounce
API._performPOSTRequest("Report_UpdateNotificationPreference", {
    reportID: announceID,
    notificationPreference: 'mute'
});
```
2. For Phones, keep E.Chat running in the background for User 1, for Web/Desktop, have the tab or window open but not focused.
3. Open up E.Chat in another client from a different account that's also in the same policy (User 2)
4. From that account send a message in the #announce room for that policy.
5. Make sure that you do not see any indicator update on the app icon for User 1

### Tested On

- [ ] Android (Not Applicable)

### Screenshots


#### Web
<img width="498" alt="Screen Shot 2021-06-25 at 4 24 15 PM" src="https://user-images.githubusercontent.com/4741899/123714816-e7dac680-d82b-11eb-912e-7d5137d7040a.png">

#### Mobile Web
N/A

#### Desktop
<img width="89" alt="Screen Shot 2021-06-28 at 4 12 17 PM" src="https://user-images.githubusercontent.com/4741899/123714793-d98caa80-d82b-11eb-94b3-8bd8ff51b0ff.png">

#### iOS
![Screen Shot 2021-06-28 at 4 55 24 PM](https://user-images.githubusercontent.com/4741899/123717628-10fe5580-d832-11eb-9031-30f0f8ea44fb.png)

#### Android
Not Applicable
